src/export/StructureDefinitionExporter.ts

```diff
@@ -1023,6 +1050,63 @@ export class StructureDefinitionExporter implements Fishable {
         }
       }
     });
+
+    // we need to double-check all our bindings in case we now contain the bound valueset.
+    // for inline instances, we should give a special error if they're not contained.
+    // for anything else, it's okay if they're not contained. but if they are, use a relative reference.
+    this.knownBindingRules.forEach((rules, sd) => {
+      for (const { rule, isInline, url } of rules) {
+        if (isInline) {
+          let vsURI = rule.valueSet;
+          const containedValueSet = sd.contained?.find((resource: any) => {
+            return resource?.id === rule.valueSet && resource.resourceType === 'ValueSet';
+          });
+          if (containedValueSet != null) {
+            vsURI = `#${containedValueSet.id}`;
+            const element = sd.findElementByPath(rule.path, this);
+            try {
+              element.bindToVS(
+                vsURI,
+                rule.strength as ElementDefinitionBindingStrength,
+                rule.sourceInfo,
+                this.fisher
+              );
+            } catch (e) {
+              logger.error(e.message, rule.sourceInfo);
+              if (e.stack) {
+                logger.debug(e.stack);
+              }
+            }
+          } else {
+            logger.error(
+              `Can not bind ${rule.path} to ValueSet ${rule.valueSet}: this ValueSet is an inline instance, but it is not present in the list of contained resources.`,
+              rule.sourceInfo
+            );
+          }
+        } else if (url) {
+          // we may have a value set with a real url that we can turn into a relative references
+          const containedValueSet = sd.contained?.find((resource: any) => {
+            return resource?.url === url;
+          });
+          if (containedValueSet != null) {
+            const element = sd.findElementByPath(rule.path, this);
+            try {
+              element.bindToVS(
+                `#${containedValueSet.id}`,
+                rule.strength as ElementDefinitionBindingStrength,
+                rule.sourceInfo,
+                this.fisher
+              );
+            } catch (e) {
+              logger.error(e.message, rule.sourceInfo);
+              if (e.stack) {
+                logger.debug(e.stack);
+              }
+            }
+          }
```

Comment:
Are you asking about the similarity between the `if` and the `else if` case? Because those seem fairly similar to me and could probably be combined by just separating out how the `containedValueSet` is calculated into an if/else.